File: src/main/java/sprites/SpriteFactory.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package sprites;

import com.jme3.texture.Texture;
import utilities.ImageManager;

/**
 *
 * @author User
 */
public class SpriteFactory {

    /**
     * Returns a Sprite based on the Image of the passed ID and the passed (x,y)
     *
     * @param ID
     * @param x
     * @param y
     * @return
     */
    public static Sprite getSprite(String ID, int x, int y, int x2, int y2) {
        try {
            return new Sprite(ImageManager.getImage(ID), x, y, x2, y2);
        } catch (NullPointerException e) {
            System.out.println("ImageManager cannot find image: " + ID);
            return null;
        }
    }

    /**
     * Returns a Sprite based on the Image of the passed ID and the passed (x,y)
     *
     * @param x
     * @param y
     * @return
     */
    public static Sprite getSprite(Texture text, int x, int y, int x2, int y2) {
        return new Sprite(text, x, y, x2, y2);
    }
    
    /**
     * Returns a Sprite based on the Image of the passed ID and the passed (x,y)
     *
     * @param ID
     * @param x
     * @param y
     * @return
     */
    /*public static Sprite getSprite(String ID, Color color, int x, int y, int x2, int y2) {
     try {
     return new Sprite(getImage(ID, color), x, y, x2, y2);
     } catch (NullPointerException e) {
     System.out.println("ImageManager.getSprite() Cannot find image: " + ID);
     return null;
     }
     }*/
}

File: src/main/java/sprites/SpriteFactory.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package sprites

import com.jme3.asset.AssetManager
import com.jme3.material.Material
import com.jme3.material.RenderState
import com.jme3.scene.Node
import com.jme3.texture.Texture2D
import com.jme3.ui.Picture

/**
 * @author brooks42
 */
class SpriteFactory(private val assetManager: AssetManager) {
    fun getSprite(text: Texture2D, x: Int, y: Int, x2: Int, y2: Int): Node {
        return constructNodeForTextureAndBounds(text, x, y, x2, y2)
    }

    private fun constructNodeForTextureAndBounds(texture: Texture2D, x: Int, y: Int, x2: Int, y2: Int): Node {
        val node = Node("test")
        val pic = getSpritePicture(texture, x2 - x, y2 - y)

        val picMat = Material(assetManager, "Common/MatDefs/Gui/Gui.j3md")
        picMat.additionalRenderState.blendMode = RenderState.BlendMode.AlphaAdditive
        node.setMaterial(picMat)

        node.attachChild(pic)
        node.move(x.toFloat(), y.toFloat(), 0f)
        return node
    }

    private fun getSpritePicture(texture: Texture2D, width: Int, height: Int): Picture {
        val pic = Picture("test")
        pic.setTexture(assetManager, texture, true)
        pic.setWidth(width.toFloat())
        pic.setHeight(height.toFloat())
        return pic
    }
}